OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -146,7 +146,7 @@ IEnumerable<Order> InnerOrder(World world, CPos cell, MouseInput mi)
 
 		public void Tick(World world)
 		{
-			if (queue.CurrentItem() == null || queue.CurrentItem().Item != actorInfo.Name)
+			if (queue.AllQueued().All(i => i.Item != actorInfo.Name))
```

Comment:
This probably also needs to check that the item has completed construction?  Otherwise this will change behaviour if the building being placed is cancelled, but another item of the same type exists later in the queue (but hasn't started construction yet)